web_poet/page_inputs.py

```diff
@@ -74,6 +75,45 @@ def from_name_value_pairs(cls: Type[T_headers], arg: List[Dict]) -> T_headers:
         """
         return cls([(pair["name"], pair["value"]) for pair in arg])
 
+    @classmethod
+    def from_bytes(
+        cls: Type[T_headers], arg: BytesDict, encoding: str = "utf-8"
+    ) -> T_headers:
+        """An alternative constructor for instantiation where the header-value
+        pairs are in raw bytes form.
+
+        This supports multiple header values in the form of ``List[bytes]``
+        alongside a plain ``bytes`` value.
+
+        By default, it converts the ``bytes`` value using "utf-8". However, this
+        can easily be overridden using the ``encoding`` parameter.
```

Comment:
This `utf-8` default value was based on how Scrapy was using it as the default for its Headers.